src/ros_filter.cpp

```diff
@@ -1974,7 +1986,13 @@ namespace RobotLocalization
       }
 
       // Fire off the position and the transform
-      positionPub_.publish(filteredPosition);
+      if (!corrected_data)
+      {
+        positionPub_.publish(filteredPosition);
+      }
+
+      // Retain the last published stamp so we can detect repeated transforms in future cycles
+      lastPublishedStamp_ = filteredPosition.header.stamp;
```

Comment:
This should be in the if statement, or else you could update based on a `corrected_data` update.

Reply:
So the earlier suggestion about >= notwithstanding, as of now, if `corrected_data == true`, then `lastPublishedStamp_ == filteredPosition.header.stamp` anyway. However, I'm perfectly happy to move this up to line 1992.